### PR TITLE
Fix so platform-specific examples show up

### PR DESCRIPTION
**Pull request by caabernathy**
After adding simulator examples, it appears that some don't show up, ex:
http://facebook.github.io/react-native/docs/text.html#content

This is due to the fact that these examples are split into two platform-specific files. For this reason, the example code isn't being shown. The examples are also missing for TextInput.

After discussions with @bestander and @mkonicek the approach taken is to list both the iOS and Android specific examples and rework the logic so the simulator is shown.

Other missing examples are Picker and Alert. These just need some minor tweaks to make sure they show up, such as renaming PickerAndroidExample.js to PickerAndroid and adding Alert to the iOS example list.

The latest apps were rebuilt uploaded to appetize.io to make sure the tests passed as some new examples had been added to the list, ex: Slider, and Picker.

Lastly, I took care of some Lint warnings, some that were unrelated to my code changes.

**Test plan (required)**

Ran the website locally. Tested to see that missing examples are shown:

- http://localhost:8079/react-native/docs/text.html#content
- http://localhost:8079/react-native/docs/textinput.html#content
- http://localhost:8079/react-native/docs/picker.html#content
- http://localhost:8079/react-native/docs/alert.html#content

I ensured the correct example was shown and that the Edit on GitHub link went to the right files for examples that had multiple files.

I also made sure the other examples were shown if necessary.

Here are some screenshots for the Text component:

screen shot 2016-04-18 at 5 30 17 pm

The iOS and Android examples are split out in this scenario:

screen shot 2016-04-18 at 5 30 35 pm

screen shot 2016-04-18 at 5 30 50 pm

### CODE REVIEW

**Comment by bestander** [comment]
Something odd with the tests.
@caabernathy can you rebase on master maybe?

**Comment by caabernathy** [comment]
Checking CircleCI logs, the failure is due to the code-analysis-bot script not being able to handle files that are renamed as part of a PR. All other tests pass.

**Comment by mkonicek** [comment]
Very cool @caabernathy!

Would it be possible to show the iOS simulator and the Android emulator below it? Would be super nice to be able to play with a cross-platform component on both platforms.

**Comment by mkonicek** [comment]
Sorry it took me a while to respond, feel free to ping me on messenger.

File: website/server/extractDocs.js [hunk 1]
```diff
@@ -54,18 +54,48 @@ function getPlatformFromPath(filepath) {
...
+}
+
+function getExamples(componentName, componentPlatform) {
+  var paths = getExamplePaths(componentName, componentPlatform);
+  if (paths) {
+    var examples = [];
+    paths.map((p) => {
+      var platform = p.match(/Example\.(.*)\.js$/);
+      var title =
+        (componentPlatform === CROSS_SUFFIX) &&
```
Comment (bestander) [inline website/server/extractDocs.js]: This form of `string = boolean && boolean && string` is a bit hard for my brain :)
I would make it an explicit assignment and not abuse coercion.

File: website/server/extractDocs.js [hunk 2]
```diff
@@ -95,14 +125,12 @@ function filterMethods(method) {
...
 
 function isRunnable(componentName, componentPlatform) {
-  var path = '../Examples/UIExplorer/' + componentName + 'Example.js';
-  if (!fs.existsSync(path)) {
-    path = '../Examples/UIExplorer/' + componentName + 'Example.' + componentPlatform + '.js';
-    if (!fs.existsSync(path)) {
-      return false;
-    }
+  var paths = getExamplePaths(componentName, componentPlatform);
+  if (paths && paths.length > 0) {
```
Comment (bestander) [inline website/server/extractDocs.js]: `return paths && paths.length > 0`

**Comment by bestander** [comment]
Great job, @caabernathy! 👍 
Just a few nits.
According to code it seems that examples for both platforms are shown one under another, right?

**Comment by bestander** [comment]
We definitely need to render docs for PRs, it is on my list :)

File: website/server/extractDocs.js (hunk 2 above)
Comment (mkonicek) [inline website/server/extractDocs.js]: Actually `undefined && foo` evaluates to undefined so seems better to explicitly return `true`, `false` just to be safe.

Comment (bestander) [inline website/server/extractDocs.js]: Good catch

**Comment by mkonicek** [comment]
@facebook-github-bot import

**Comment by facebook-github-bot-8** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1689548384645046/int_phab) to review.

**Comment by mkonicek** [comment]
Rebased on top of master: https://github.com/facebook/react-native/pull/7406

@caabernathy You get attribution (git blame) correctly.

**Pull request closed without merging**